pkg/controller/plan/kubevirt.go

```diff
@@ -1835,6 +1848,156 @@ func (r *KubeVirt) EnsurePersistentVolume(vmRef ref.Ref, persistentVolumes []cor
+	err = client.Get(
+		context.TODO(),
+		types.NamespacedName{
+			Name:      fmt.Sprintf("ova-nfs-pvc-%s-%s", providerName, planName),
+			Namespace: planNamespace,
+		},
+		pvc,
+	)
+
+	if err != nil {
+		if k8serr.IsNotFound(err) {
+			return nil, false, nil
+		}
+		err = liberr.Wrap(err)
+		return
+	}
+	return pvc, true, nil
+}
+
+// TODO if we use ownership get it to the method
+func (r *KubeVirt) CreatePvForNfs() (err error) {
+	sourceProvider := r.Source.Provider
+	splitted := strings.Split(sourceProvider.Spec.URL, ":")
+	if len(splitted) != 2 {
+		r.Log.Error(nil, "NFS server path doesn't contains: ", "url", sourceProvider.Spec.URL)
+		return fmt.Errorf("bad source provider %s URL %s", sourceProvider.Name, sourceProvider.Spec.URL)
+	}
+	nfsServer := splitted[0]
+	nfsPath := splitted[1]
+
+	pvName := fmt.Sprintf("ova-nfs-pv-%s-%s", r.Source.Provider.Name, r.Plan.Name)
+	_, found, err := GetOvaPvNfs(r.Destination.Client, r.Plan.Name, r.Plan.Provider.Source.Name)
+	if err != nil {
+		return
+	}
+	if found {
+		r.Log.Info("The PV for OVA NFS exists", "PV", pvName)
+		return
+	}
+
+	pv := &core.PersistentVolume{
+		ObjectMeta: meta.ObjectMeta{
+			Name: pvName,
+		},
+		Spec: core.PersistentVolumeSpec{
+			Capacity: core.ResourceList{
+				core.ResourceStorage: resource.MustParse("1Gi"),
+			},
+			AccessModes: []core.PersistentVolumeAccessMode{
+				core.ReadOnlyMany,
+			},
+			PersistentVolumeSource: core.PersistentVolumeSource{
+				NFS: &core.NFSVolumeSource{
+					Path:   nfsPath,
+					Server: nfsServer,
+				},
+			},
+		},
+	}
+	err = r.Destination.Create(context.TODO(), pv)
+	if err != nil {
+		r.Log.Error(err, "Failed to create OVA plan PV")
+		return
+	}
+	return
+}
+
+func (r *KubeVirt) CreatePvcForNfs(pvcName string) (err error) {
+	_, found, err := GetOvaPvcNfs(r.Destination.Client, r.Plan.Name, r.Plan.Namespace, r.Plan.Provider.Source.Name)
+	if err != nil {
+		return
+	}
+	if found {
+		r.Log.Info("The PVC for OVA NFS exists", "PVC", pvcName)
+		return
+	}
+
+	sc := ""
+	pvName := fmt.Sprintf("ova-nfs-pv-%s-%s", r.Source.Provider.Name, r.Plan.Name)
+	pvc := &core.PersistentVolumeClaim{
+		ObjectMeta: meta.ObjectMeta{
+			Name:      pvcName,
+			Namespace: r.Plan.Spec.TargetNamespace,
+			// OwnerReferences: []meta.OwnerReference{ownerReference},
+		},
+		Spec: core.PersistentVolumeClaimSpec{
+			Resources: core.ResourceRequirements{
+				Requests: core.ResourceList{
+					core.ResourceStorage: resource.MustParse("1Gi"),
+				},
+			},
+			AccessModes: []core.PersistentVolumeAccessMode{
+				core.ReadOnlyMany,
+			},
+			VolumeName:       pvName,
+			StorageClassName: &sc,
+		},
+	}
+	err = r.Destination.Create(context.TODO(), pvc)
+	if err != nil {
+		r.Log.Error(err, "Failed to create OVA plan PVC")
+		return
+	}
+
+	// wait until pvc and pv are bounded.
+	timeout := time.After(5 * time.Minute)
+	tick := time.Tick(5 * time.Second)
```

Comment:
golintci has a point

Reply:
yeah, the `wait` can be a good candidate here